charts/dapr/crds/pluggable-components.yaml

```diff
@@ -15,36 +16,38 @@ spec:
     singular: pluggablecomponent
   scope: Namespaced
   versions:
-    - name: v1alpha1
-      schema:
-        openAPIV3Schema:
-          description: PluggableComponent is a especification for a pluggable component.
-          properties:
-            apiVersion:
-              description:
-                "APIVersion defines the versioned schema of this representation
-                of an object. Servers should convert recognized schemas to the latest
-                internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources"
-              type: string
-            kind:
-              description:
-                "Kind is a string value representing the REST resource this
-                object represents. Servers may infer this from the endpoint the client
-                submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds"
-              type: string
-            metadata:
-              type: object
-            spec:
-              description: PluggableComponentSpec is the spec for a pluggable component.
-              properties:
-                type:
-                  type: string
-                version:
-                  type: string
-              required:
-                - type
-                - version
-              type: object
-          type: object
-      served: true
-      storage: true
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: PluggableComponent is a especification for a pluggable component.
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: PluggableComponentSpec is the spec for a pluggable component.
+            properties:
+              name:
+                description: Name is the component name if not specified metadata.Name
```

Comment:
I was considering use array of types instead to have a normalized experience. On the other hand, it can be harder to extend later since it would need to be an array of tuples (not just `type`).

The best experience IMO would be to rename `name` to `componentName` - this way, it is clear that is not the k8s resource name but the componentName referenced in the component CRD. So, users could have a pluggable component that implements multiple component types with the following PluggableComponent resources in K8s:

(resource) name | metadata.type | metadata.version | metadata.componentName
-|-|-|-
myredis-state | state | 1 | myredis
myredis-inputbinding | inputbinding | 1 | myredis
myredis-outputbinding | outputbinding | 1 | myredis

Then in the component CRD, user would just need to reference `myredis` as a state, inputbinding or outputbinding.

Reply:
Yes, I agree with you. Applied your suggestions thanks @artursouza